env0/resource_project_policy.go

```diff
@@ -165,14 +171,18 @@ func resourcePolicyUpdate(ctx context.Context, d *schema.ResourceData, meta inte
 		payload.MaxTtl = ""
 	}
 
+	if payload.DriftDetectionCron != "" {
+		payload.DriftDetectionEnabled = true
+	}
+
 	if _, err := apiClient.PolicyUpdate(payload); err != nil {
 		return diag.Errorf("could not update policy: %v", err)
 	}
 
 	return nil
 }
 
-func resourcePolicyDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+func resourceProjectPolicyDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
Shouldn't we disable the cron in case the project policy is deleted?

Reply:
I don't know... wouldn't it happen anyway if the policy is removed? How would it be able to run a cron for a policy that doesn't exist?

Reply:
what happens when a policy is removed?
I don't think we have an endpoint to "delete a policy"

Reply:
We don't
The policy should be reset using the same API (that's what the UI do, I think - it sends an empty value)

Reply:
@yaronya that's a bad idea it means this client will need to know what are the default values of each key.

Why don't we add an endpoint to reset the policy to default?
The backend has `getDefaultPolicies` it would simple and maintainable for the future.

Reply:
So @TomerHeber we'll take that, but I think in the mean time can we make it so the destroy sends a 'cron disabled' property, and handle that in a different ticket, sounds right?

Reply:
@avnerenv0 - yes. I'll make the required changes. Thanks!

Reply:
@avnerenv0 I have validated. No code changes are required.
On a delete, the update payload will be sent with `driftDetectionCron=""` and `driftDetectionEnabled=false`

```
type PolicyUpdatePayload struct {
	ProjectId                   string `json:"projectId"`
	NumberOfEnvironments        int    `json:"numberOfEnvironments,omitempty"`
	NumberOfEnvironmentsTotal   int    `json:"numberOfEnvironmentsTotal,omitempty"`
	RequiresApprovalDefault     bool   `json:"requiresApprovalDefault"`
	IncludeCostEstimation       bool   `json:"includeCostEstimation"`
	SkipApplyWhenPlanIsEmpty    bool   `json:"skipApplyWhenPlanIsEmpty"`
	DisableDestroyEnvironments  bool   `json:"disableDestroyEnvironments"`
	SkipRedundantDeployments    bool   `json:"skipRedundantDeployments"`
	RunPullRequestPlanDefault   bool   `json:"runPullRequestPlanDefault"`
	ContinuousDeploymentDefault bool   `json:"continuousDeploymentDefault"`
	MaxTtl                      string `json:"maxTtl,omitempty"`
	DefaultTtl                  string `json:"defaultTtl,omitempty"`
	ForceRemoteBackend          bool   `json:"forceRemoteBackend"`
	DriftDetectionCron          string `json:"driftDetectionCron"`
	DriftDetectionEnabled       bool   `json:"driftDetectionEnabled"`
}
```